netsize/netsize.go

```diff
@@ -0,0 +1,268 @@
+package netsize
+
+import (
+	"fmt"
+	"math"
+	"math/big"
+	"sort"
+	"strings"
+	"sync"
+	"time"
+
+	logging "github.com/ipfs/go-log"
+	kbucket "github.com/libp2p/go-libp2p-kbucket"
+	"github.com/libp2p/go-libp2p/core/peer"
+	ks "github.com/whyrusleeping/go-keyspace"
+)
+
+var (
+	ErrNotEnoughData   = fmt.Errorf("not enough data")
+	ErrWrongNumOfPeers = fmt.Errorf("expected bucket size number of peers")
+)
+
+var (
+	logger                   = logging.Logger("dht/netsize")
+	MaxMeasurementAge        = 2 * time.Hour
+	MinMeasurementsThreshold = 5
+	MaxMeasurementsThreshold = 150
+	keyspaceMaxInt, _        = new(big.Int).SetString(strings.Repeat("1", 256), 2)
+	keyspaceMaxFloat         = new(big.Float).SetInt(keyspaceMaxInt)
+)
+
+type Estimator struct {
+	localID    kbucket.ID
+	rt         *kbucket.RoutingTable
+	bucketSize int
+
+	measurementsLk sync.RWMutex
+	measurements   map[int][]measurement
```

Comment:
Same remark about `sync.Map`, may worth it, maybe not.

Reply:
See comment above